src/jdk.sctp/unix/classes/sun/nio/ch/sctp/SctpChannelImpl.java

```diff
@@ -639,6 +639,7 @@ public void kill() throws IOException {
             if (state == ChannelState.KILLED)
                 return;
             if (state == ChannelState.UNINITIALIZED) {
+                SctpNet.close(fdVal);
                 state = ChannelState.KILLED;
```

Comment:
It might be better to invert these, meaning set the state to KILL before close(fdVal), just in case the close throws.

Reply:
So modified.

Reply:
It looks like you've changed some but not all cases. I realize ChannelState.UNINITIALIZED is not too interesting but further maintainers may wonder about it.